bin/install-qa-check.d/60pkgconfig

```diff
@@ -1,11 +1,105 @@
+	f=$(egrep -zsH '^Libs.*-Wl,(-O[012]|--hash-style)' "${files[@]}")
 	if [[ -n ${f} ]] ; then
 		eqawarn "QA Notice: pkg-config files with wrong LDFLAGS detected:"
-		eqawarn "${f//${D}}"
+		eqatag -v pkgconfig.bad-ldlags "${f//${D}}"
+	fi
+
+	# Validate using pkgconfig
+	# Some less common implementations may not support this?
+	# seems like f.d.o, OpenBSD, and of course pkgconf do though.
+	# Need --maximum-traverse-depth=1 to avoid checking deps and giving
+	# unrelated warnings/errors.
+	if ! pkg-config --maximum-traverse-depth=1 --validate "${files[@]}" ; then
+		eqawarn "QA Notice: pkg-config files which fail validation found!"
+		eqawarn "Run 'pkg-config --validate ...' for more information"
+	fi
+
+	# Check for unexpected paths
+	# e.g. https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=c90ab38e3577aae61fac2341b34ad593948de1cd
+	if [[ -n ${EPREFIX} ]] ; then
+		for f in "${files[@]}" ; do
+			local key
+			for key in prefix exec_prefix libdir includedir ; do
+				local value=$(pkg-config --variable="${key}" "${f}")
+
+				if [[ ${value} != "${EPREFIX}"* ]] ; then
+					eqawarn "QA Notice: pkg-config files not respecting EPREFIX found"
+					eqawarn "${f}'s key=${key} does not respect EPREFIX:"
+					eqawarn "${key}=${value}"
+					eqatag -v pkgconfig.bad-paths "${f}"
+				fi
+			done
+		done
+	fi
+
+	# TODO: Generalise for non-lib64 libdir? Not that this is very common now
+	# that riscv chose a more standard layout.
+	#
+	# If we're installing to ${ED}/usr/lib/pkgconfig, let's make sure
+	# we're not referencing lib64.
+	#
+	# e.g. https://bugs.gentoo.org/729642
+	local bad_libdir=()
+	for f in "${files[@]}" ; do
+		# In ${ED}/usr/lib, we shouldn't reference lib64
+		if [[ ${f} == *lib/pkgconfig* ]] ; then
+			if [[ -d "${ED}"/usr/lib && -L "${ED}"/usr/lib ]] ; then
+				# (Don't bother if /usr/lib is a symlink to /usr/lib64)
+				continue
+			fi
+
+			if egrep -q "/lib64" ${f} ; then
+				bad_libdir+=( ${f} )
+			fi
+		elif [[ ${f} == *lib64/pkgconfig* ]] ; then
+			# We want to match /lib/, /lib/foo/, but not e.g. /lib64 or /lib64/, or libfoo
+			if grep -qP '/lib\b' ${f} ; then
+				bad_libdir+=( ${f} )
+			fi
+		fi
+	done
+
+	if [[ -n "${bad_libdir[@]}" ]] ; then
+		eqawarn "QA Notice: pkg-config files not respecting libdir found"
+		eqawarn "(contains reference to either lib or lib64 in wrong directory)"
+		eqatag -v pkgconfig.bad-libdir "${bad_libdir[@]}"
+	fi
+
+	# Check for mismatched Version field vs ${PV}
+	# To be safe, let's make sure _all_ installed .pcs have a bad Version
+	# before warning, as this should catch the general cases we're worried
+	# about, while avoiding any pathological cases e.g. multiple libraries
+	# with different versioning within one package.
+	# Example bugs: bug #833895, bug #833887.
+	local all_bad=yes
+	# Record the last bad file matched
+	local bad_file
+	for f in "${files[@]}" ; do
+		if [[ ${PV} == $(sed -n -e 's/^Version: //p' "${f}" || die) ]] ; then
```

Comment:
Any reason you're calling pkg-config in the above checks but working on the file directly here?

Reply:
In some cases, the version has been a ridiculous string ("This version was generated by a git tar ball") or something and I wasn't convinced pkg-config would handle it correctly/notice. But I don't mind changing and experimenting.

Reply:
Well, it seems to truncate it to the first word but I suppose that may be good enough. Your choice.

```
# pkg-config --modversion libical
yo
```

Reply:
I'm going to stick to my original so that I can notice something like "1.2.0 (git)" or similar.